core/src/main/java/org/apache/carbondata/core/datamap/DataMapUtil.java

```diff
@@ -0,0 +1,132 @@
+import org.apache.carbondata.core.indexstore.ExtendedBlocklet;
+import org.apache.carbondata.core.indexstore.PartitionSpec;
+import org.apache.carbondata.core.indexstore.blockletindex.BlockletDataMapFactory;
+import org.apache.carbondata.core.metadata.schema.table.CarbonTable;
+import org.apache.carbondata.core.scan.filter.resolver.FilterResolverIntf;
+import org.apache.carbondata.core.statusmanager.SegmentStatusManager;
+import org.apache.carbondata.core.util.ObjectSerializationUtil;
+
+import org.apache.hadoop.conf.Configuration;
+
+public class DataMapUtil {
+
+  private static final String DATA_MAP_DSTR = "mapreduce.input.carboninputformat.datamapdstr";
+
+  private static final LogService LOGGER =
+      LogServiceFactory.getLogService(DataMapUtil.class.getName());
+
+  /**
+   * This method set DataMapJob if configured
+   *
+   * @param conf
+   * @throws IOException
+   */
+  public static void setDataMapJobIfConfigured(Configuration conf) throws IOException {
+    String className = "org.apache.carbondata.spark.rdd.SparkDataMapJob";
+    setDataMapJob(conf, createDataMapJob(className));
+  }
+
+  /**
+   * Creates instance for the DataMap Job class
+   *
+   * @param className
+   * @return
+   */
+  public static Object createDataMapJob(String className) {
+    try {
+      return Class.forName(className).getDeclaredConstructors()[0].newInstance();
+    } catch (Exception e) {
+      LOGGER.error(e);
+      return null;
+    }
+  }
+
+  public static void setDataMapJob(Configuration configuration, Object dataMapJob)
+      throws IOException {
+    if (dataMapJob != null) {
+      String toString = ObjectSerializationUtil.convertObjectToString(dataMapJob);
+      configuration.set(DATA_MAP_DSTR, toString);
+    }
+  }
+
+  public static DataMapJob getDataMapJob(Configuration configuration) throws IOException {
+    String jobString = configuration.get(DATA_MAP_DSTR);
+    if (jobString != null) {
+      return (DataMapJob) ObjectSerializationUtil.convertStringToObject(jobString);
+    }
+    return null;
+  }
+
+  public static void executeDataMapJobForClearingDataMaps(CarbonTable carbonTable)
+      throws IOException {
+    String dataMapJobClassName = "org.apache.carbondata.spark.rdd.SparkDataMapJob";
+    DataMapJob dataMapJob = (DataMapJob) createDataMapJob(dataMapJobClassName);
+    String className = "org.apache.carbondata.core.datamap.DistributableDataMapFormat";
+    SegmentStatusManager ssm = new SegmentStatusManager(carbonTable.getAbsoluteTableIdentifier());
+    List<Segment> validSegments = ssm.getValidAndInvalidSegments().getValidSegments();
+    DataMapExprWrapper dataMapExprWrapper = null;
+    if (DataMapStoreManager.getInstance().getAllDataMap(carbonTable).size() > 0) {
+      DataMapChooser dataMapChooser = new DataMapChooser(carbonTable);
+      dataMapExprWrapper = dataMapChooser.getAllDataMapsForClear(carbonTable);
+    } else {
+      return;
+    }
+    DistributableDataMapFormat dataMapFormat =
```

Comment:
What if `dataMapExprWrapper` is null?

Reply:
dataMapExprWrapper  will be null, if the table does not have datamaps, that check is already there

Reply:
So won't it fai in line 100, if you don't do null check?

Reply:
no, if there are no datamaps present, it will return